src/Label.ml

```diff
@@ -0,0 +1,100 @@
+open Ast
+
+type block =
+        | BlAssign of string * a
+        | BlExpBool of b
+        | BlSkip
+        [@@deriving show]
+
+module Label =
+        struct
+                type t = int
+                let compare l_a l_b = Stdlib.compare l_a l_b
+        end
+
+module Edge =
+        struct
+                type t = int * int [@@deriving ord]
+                let compare e_a e_b = compare e_a e_b
+        end
+
+module LabelMap = Map.Make(Label)
+module LabelSet = Set.Make(Label)
+
+let rec blocks_of stm blocks = match stm with
+        | Assign(s, a, l) -> LabelMap.add l (BlAssign (s, a)) blocks 
+        | Seq(s_1, s_2) -> 
+                let bl_1 = blocks_of s_1 blocks in
+                blocks_of s_2 bl_1
+        | Skip(l) -> LabelMap.add l BlSkip blocks
+        | Ifte(b, s_1, s_2, l) ->
+                let bl_b = LabelMap.add l (BlExpBool b) blocks in
+                let bl_1 = blocks_of s_1 bl_b in
+                blocks_of s_2 bl_1
+        | While(b, s, l) ->
+                let bl_b = LabelMap.add l (BlExpBool b) blocks in
+                blocks_of s bl_b                
+
+let labels stm =
+        let rec acc lb_list = function
```

Comment:
`acc` is traditionally reserved for *acc*umulators. Use `go` instead, or `help`, for an anonymous recursive function.